server/events/vcs/github_client.go

```diff
@@ -451,6 +451,8 @@ func (g *GithubClient) GetCombinedStatusMinusApply(repo models.Repo, pull *githu
 				}
 
 			}
+		} else {
+			return false, nil
```

Comment:
This is not always correct because a check can be in progress but it may as well be not set as required, in which case PR will still be mergeable.

Reply:
This is checked prior to the function call here: https://github.com/runatlantis/atlantis/blob/8f97dedf188863669b85f9fbcf4b54240c65e16c/server/events/vcs/github_client.go#L497-L534

The state you are referring to is considered "unstable"

Reply:
I think @stasostrovskyi is right here. Assume `atlantis/apply` is pending and one another check which is not a required is also pending. So the status will not be `unstable` and will go on to the function `GetCombinedStatusMinusApply` since `AllowMergeableBypassApply` is enabled. 

There the changes that are done in this PR will come into picture. Here it will fail the check `if *c.Status == "completed"` and proceed to  to return `return false, nil`, even before checking if the status check is failing due to a required check or not.

I think we might have to go further if `*c.Status != "completed"` and then check if the checkrun is a required one or not.

Reply:
For me think it _finally_ makes sense with the issue you've discovered and fixing in https://github.com/runatlantis/atlantis/pull/3916